### PR TITLE
Setup Travis and basic file structure

### PR DESCRIPTION
**Pull request by rwblickhan**
I created a `main` function as well as a basic directory structure, and also set up Travis so that it can build on every pull request (this will however need to be activated on [travis-ci.org](travis-ci.org)). Also I set the C++ standard to C++14 for the time being; it doesn't change much from C++11 (i.e. modern C++) but since we don't really have to worry about backwards compatibility and cross-platform and so on, we might as well use the most recent version.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+cmake_minimum_required(VERSION 3.0)
+project(durian)
+
+set(CMAKE_CXX_STANDARD 14)
+
+include_directories(${PROJECT_SOURCE_DIR}/src/include)
```
Comment (bobheadxi) [inline CMakeLists.txt:6]: Shouldn't we `.gitignore` the cmake output?

**Review by bobheadxi — APPROVED** [review]
Looks great! 🔥 Only thing that might need changing is a `.gitignore`

```cpp
#include <stdio.h>

int main() {
    printf("Hello world!\n");
    return 0;
}
```

```bash
bobbook:durian robertlin$ cmake CMakeLists.txt
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/robertlin/Projects/durian
bobbook:durian robertlin$ make
Scanning dependencies of target durian
[ 50%] Building CXX object CMakeFiles/durian.dir/src/main.cpp.o
[100%] Linking CXX executable durian
[100%] Built target durian
bobbook:durian robertlin$ ./durian
Hello world!
```

**Review by bwdmonkey — COMMENTED** [review]
LGTM. I assume we put the header files inside `src/include/` and the `.cpp` files in the `src/`?

File: CMakeLists.txt (hunk 1 above)
Comment (bwdmonkey) [reply CMakeLists.txt:6]: I'd include the executables after `make` is ran as well.

Comment (rwblickhan) [reply CMakeLists.txt:6]: Yeah, we should. I was also going to include IDE files but wanted by to ask around what IDEs y’all were planning to use first.

Comment (bobheadxi) [reply CMakeLists.txt:6]: VScode is life sir

Comment (bwdmonkey) [reply CMakeLists.txt:6]: VSCode

Comment (rwblickhan) [reply CMakeLists.txt:6]: Do you guys happen to know what IDE files VScode uses?

Comment (bobheadxi) [reply CMakeLists.txt:6]: vscode config is in `.vscode/settings.json` 

Standard setup procedure: https://code.visualstudio.com/docs/languages/cpp

**Comment by rwblickhan** [comment]
Yep, `.h` goes into `src/include/` and `.cpp` goes into `src/`, or at least that's the intention. Presumably we would also add subdirectories of `src/` for each major component (i.e. `src/lexer/`, `src/parser/`, etc), and *maybe* mirror that in `src/include/`.

But that's just one way of doing things (albeit one I prefer 😛) and we could discuss other ways of structuring the project.

**Comment by rwblickhan** [comment]
I've added the .gitignore. I also added a section on building to the README which explains how to get it to build if you want to do it from the command line (which is useful for working with Travis, for instance).

**Pull request merged**